### ISSUE TITLE
Legacy project support for ImplicitlyDefined/AutoReference PackageReferences not working.

### ISSUE THREAD

**Issue by sae42** [NuGet/Home#7319]
## Package Manager UI does not honor AutoReference/ImplicitlyDefined package references
VS version (if appropriate):
Using VS 2017 15.8.4 (or VS15.9 Preview)

OS version (i.e. win10 v1607 (14393.321)):
win10 v1803

1. When using an AutoReference PackageReference in a legacy C# project (e.g. Windows Forms), the package manager UI still allows updating. E.g. in the csproj add something like:
    <PackageReference Include="EntityFramework">
      <Version>6.1.0</Version>
      <IsImplicitlyDefined>True</IsImplicitlyDefined>
    </PackageReference>

2. In VS, open the C# project and "Manage NuGet Packages...". The reference is allowed to be updated.

This is a contrived example; I'm looking for the ability to block updating a package reference in an imported .targets much like you get with Microsoft.NETCore.App in a .NET Core project.

I'm not sure if this feature should be available in legacy projects, but it would be nice if it was. My code inspection of the NuGet client tooling leads me to suspect the issue is rooted in PackageReference types having to be a 'BuildIntegratedPackageReference' - e.g. in DetailControlModel.cs (SetCurrentPackage etc).

**Comment by jainaashish** [comment]
If I'm not wrong you're asking for `AutoReference` like feature for non-SDK styled projects? And the purpose of this feature is to restrict updates through UI.. right?

If that's the ask then you can set the version property to be a closed range on PackageReference in legacy csproj which will achieve the similar experience for you. It will restrict NuGet updates to that closed range. Now your closed range can be just one version or multiple... depending on your requirement. Does this help? if not, then please explain your scenario in detail.

**Comment by sae42** [comment]
Yes, that's right - though it's more to do with the project system implementing VSProject4 PackageReference support. I've used legacy C# as the example, but I actually have my own project system (non-CPS) which I'm attempting to add PackageReference support to. NuGet is setting the AutoReference but the UI which handles the updating can't handle it.

I tried experimenting with closed versions but this didn't seem to work either. The package manager UI would happily update/modify a PackageReference.

**Comment by jainaashish** [comment]
`I tried experimenting with closed versions but this didn't seem to work either. The package manager UI would happily update/modify a PackageReference.`

Can you repro it with legacy c# project as well? if yes, then give us the repro steps or repro solution to further investigate it.

Ideally AutoReference would have also worked for you, but lets start with closed range n then we can also look into autoReference issue.

**Comment by sae42** [comment]
Yes, I can repro with a legacy C# project. It's quite straightforward:
1. Simply create a new C# Windows Forms app
2. Manage NuGet Packages and add something. Let's use Newtonsoft.Json. Pick an older version (e.g. 10.0.1).
3. If you unload the csproj and look you'll have:
    <PackageReference Include="Newtonsoft.Json">
      <Version>10.0.1</Version>
    </PackageReference>
4. Change the version to be [10.0.1] and reload.
5. In Manage NuGet packages you can still update, remove and see a list of all versions etc. If you update, the <Version> will be rewritten back to the csproj as something like <Version>11.0.2</Version>. So it looks like this is being completely ignored too.

**Comment by jainaashish** [comment]
ahh i see, it doesn't show updates available for that package, but still shows all the versions in Installed tab and allowed to update to any latest version.

**Comment by rrelyea** [comment]
@jainaashish - is this really a regression? when did this work?

**Comment by jainaashish** [comment]
I distinctly remember this working when we added support of blocking packages update in UI... Now i'm not sure when did we add that support and when we broke it...
